### DISCUSSION TITLE
Is there a documentation of the docker-compose file?

### DISCUSSION THREAD

**Discussion by mikedolx** [FraunhoferIOSB/FROST-Server#1310] [Q&A] [answered]
Hi,

i was wondering, if there is any documentation of the docker-compose file especially the environment variables?

My questions regarding the docker-compose would be:
 - Actually, i was wondering, what the difference is between the `persistence_db_*` settings and the `auth_db_*` settings? 
 - How can i provide custom authentication to the mqtt-part, if i have set up my broker with specific user and password?

**Comment by hylkevds** [comment, marked as answer]
All documentation is on the [documentation site](https://fraunhoferiosb.github.io/FROST-Server/), including all [settings](https://fraunhoferiosb.github.io/FROST-Server/settings/settings.html).

The reason that the Basic Auth module has its own database settings is so that the database containing the auth* information can be different from the database containing the data.

I don't understand your last question. There is no "my broker", FROST-Server is the broker as shown in the [architecture](https://fraunhoferiosb.github.io/FROST-Server/deployment/architecture-packages.html).

**Reply by mikedolx** [reply]
I think it's the message bus, that we are talking about.

In [this docker-compose](https://github.com/FraunhoferIOSB/FROST-Server/blob/v2.x/scripts/docker-compose-separated.yaml), there is a mosquitto-broker defined. From the settings i can provide to the mqtt-part, i assume, that i am free to choose the broker that is used as the message broker?

**Reply by hylkevds** [reply]
Yes, the message bus can be any MQTT broker.
Setting credentials for that connection is currently not supported...

**Reply by mikedolx** [reply]
OK, thanks. Is it somehow possible to define the clientId, that the FROST-MQTT client uses towards the broker?

**Reply by hylkevds** [reply]
The clientId is currently fixed to `"FROST-MQTT-Bus-" + UUID.randomUUID()` and not configurable.
